### DISCUSSION TITLE
FAQs on Cronos POS, Cronos EVM & Cronos zkEVM Chains

### DISCUSSION THREAD

**Discussion by aw126** [crypto-org-chain/chain-main#442] [Q&A]
## What is the difference among Cronos POS, Cronos EVM and Cronos zkEVM Chains?

The Cronos ecosystem is now multi-chain across several networks in order to support a broad range of user needs and use cases:

- Cronos POS chain is the layer-0 network that guarantees the security of the Cronos ecosystem. It is a public, open-source and permissionless chain with high speed and low fees, designed to help drive mass adoption of blockchain technology through use cases like Payments and NFTs. 
The network is supported by world-class ecosystem partners including Crypto.com, Ledger, and Coinbase, and is maintained by over 100 active validators with a multi-year track record of secure and reliable performance. It also serves as the native issuer of the $CRO utility token.

- Cronos EVM chain is the leading Ethereum-compatible network built on the Cosmos SDK, permissionless, programmable layer-1, home to more than 500 project teams who have created decentralized applications powered by smart contracts. 
Cronos is extendable, thanks to application-specific optimistic layer-2 blockchains that can be deployed on top of Cronos by app creators. It remains the preferred network for community-based use cases around Web3 games and NFTs.

- Cronos zkEVM chain is the latest sister chain that aims to extend the rich portfolio of Cronos apps into the promising zkSync hyperchain ecosystem, in order to further broaden its user base and make it easier to move liquidity back and forth between Ethereum and the Cronos chain(s). It is particularly well suited for use cases that benefit from deep liquidity and integration with Ethereum, such as DeFi, trading, and cross-chain decentralized applications.

### What is the consensus algorithm used by Cronos POS Chain?
We adopt Tendermint Core as the consensus engine of the Cronos POS Chain. Tendermint is one of the most mature Byzantine-fault tolerant (BFT) consensus engines for building Proof-of-stake (PoS) systems. 

### What are the rewards for staking CRO?
The staking APR (Annual Percentage Rate) is dynamic and may fluctuate based on network conditions such as the total amount staked and validator commission rates. You can check the latest APR on [Mintscan](https://www.mintscan.io/crypto-org), which reflects the real-time return rate for stakers, as of now, it's approximately 7%.
Rewards are generated every block, so stakers begin earning continuously once their CRO is delegated. In addition to these rewards, validators may earn commission from their delegators, which is set individually by each validator.
 
### How should I stake my CRO?
CRO can be staked directly on the Cronos POS Chain using wallets like the [Crypto.com Onchain Wallet](https://crypto.com/en/defi/wallet/) to perform CRO staking. Also, participants can also perform staking by [Crypto.com Defi Desktop Wallet](https://github.com/crypto-com/chain-desktop-wallet/releases/). The [Delegation Guide](https://docs.cronos-pos.org/for-users/wallets/delegation-guide) is designed to provide you with clear instructions for managing your CRO stakes across different platforms. 
Other than these, participants are always welcome to use [chain-maind CLI](https://docs.cronos-pos.org/cronos-pos-chain-protocol/module_overview/module_staking#transactions-and-queries) to perform staking. 
 
### Where can I check the details about validators?
For information about validators, use the [Cronos POS explorer](https://cronos-pos.org/explorer/validators). The explorer's search feature lets you look up any validator by entering a block, transaction, account, or validator address. 
Users can also refer to some 3rd parties explorer like [Mintscan](https://www.mintscan.io/crypto-org/validators), [Validator.info](https://validator.info/cronos-pos) and [Defier](https://defier.net/) for the insight about validators. 
 
### Can I choose the validator to delegate my tokens?
Yes, you can choose your preferred validator when staking CRO tokens using the Crypto.com DeFi Desktop Wallet or the Crypto.com DeFi Wallet mobile app. This allows you to delegate to validators based on your preferences for performance, commission rate, or other factors.
 
### Is there a limited number of validators?
Yes, there's a cap on "active" validators. The mainnet supports a maximum of 100 active validators, with only the top 100 validators ranked by total staking amount qualifying as "active" and eligible to earn rewards.

### What is the minimum staking required to be a validator?
There is no minimum stake amount to delegate, and you can always start with 1 CRO.

### What is the requirement for being a validator?
For Cronos POS mainnet, you will need a machine with the following minimum requirements to run different types of nodes:
Archive Node (setting pruning = nothing)
- RAM: 64GB (Rocksdb)
- Disk: 5.9TB
- CPU: 4 cores

Default Full Node (setting pruning = default)
- RAM: 64GB (Rocksdb) or 16GB (goleveldb)
- Disk: 1.5TB (From quick sync)
- CPU: 4 cores

Pruned Node (setting pruning = everything)
- RAM: 64GB (Rocksdb) or 16GB (goleveldb)
- Disk: 15GB (From Publicnode)
- CPU: 4 cores

Refer to [this page](https://docs.cronos-pos.org/for-node-hosts/getting-started/mainnet#prepare-your-machine) for the latest details. 

### What is the benefit of being the top 100 validator node?
The top 100 validators are considered to be "Active" and earn rewards.

### What is the difference between a full node, a validator, and a state-sync node? 
A full node maintains the complete transaction history of the blockchain. Running a full node gives you access to all historical chain data but requires substantial storage space due to the comprehensive database.
A validator node is a specialized node that verifies transactions and signs blocks. Validators play a critical role in network security by validating new transactions before they're added to the blockchain.
A state-sync node uses state synchronization to quickly sync with the network without downloading the entire blockchain history. This type of node doesn't store the full blockchain database, making it more storage-efficient while still allowing participation in the network.

### What is the difference between a full node and a sentry node? 
They are similar, while a sentry node has one or more private peers, and these peers may be validators or other full nodes in the network. A sentry node is meant to provide a layer of security for your validator. 

### What does the 28-day unbonding period mean?
When you unstake your CRO tokens, there's a mandatory 28-day waiting period before you can access them again. This is to protect against a validator attacking the Cronos POS Chain and then immediately withdrawing the stake amount. 

### Is there a time limit for redelegating? or how often can I redelegate to a new validator? 
You can redelegate CRO once every 28 days. Once the tokens enter unbonding period, you will have to wait for 28 days before doing any other operation including redelegating.

### Under what circumstances/conditions can a validator become jailed?
Punishments for a validator are triggered when they either make a byzantine fault or become non-live:
#### Liveness Faults (Low availability)
A validator is said to be non-live when they fail to sign at least 2500 blocks (around 4.2 hrs) in the last 5000 blocks in the mainnet.
#### Byzantine Faults
A validator is said to make a byzantine fault when they sign conflicting messages/blocks at the same height and round. 
Tendermint has mechanisms to publish evidence of validators that signed conflicting votes so they can be punished by the slashing module. For example:
- Validator who votes for two different blocks within a single round;
- Validator who signs commit messages for arbitrary application states.

### Will my (user) stake/delegation be slashed immediately after a validator becomes jailed?
For liveness faults (e.g. downtime or missing blocks), validators are jailed but no slashing is applied. 
For byzantine faults (e.g. double signing), validators are jailed and 5% of the validator’s and delegators’ stake will be slashed immediately.

### What happens if the validator is unbonded or jailed? 
You will not lose your CRO if your validator becomes unbonded or jailed. However:
- While the validator is jailed, it cannot earn rewards until it is unjailed.
- If the validator is unbonded, it is removed from the active set and no longer participates in consensus or earns rewards.
If you have not redelegated in the past 28 days, you can still delegate to an active validator.

### How can I check if my node (validator) status has missed any blocks? 
You can check the information by command:
`chain-maind q slashing signing-info <crocnclconspub address of validator>`

### How can I change the commission rate?
`chain-maind tx staking edit-validator --from [wallet] --commission-rate="[rate]" --chain-id "[chain]" --gas-prices 0.1basecro`

### How can I withdraw commissions from the validator? 
`chain-maind tx distribution withdraw-rewards [delegator addrs] --from [key] --chain-id "crypto-org-chain-mainnet-1" --gas-prices 0.1basecro --commission`

### Is there a way to change the details of my validator?
Yes, you can make changes with the "edit_validator" command.
`chain-maind tx staking edit-validator --help`

### How can I withdraw rewards from delegation?
`chain-maind tx distribution withdraw-rewards [delagator] --from [key] --chain-id "crypto-org-chain-mainnet-1" --gas-prices 0.1basecro --commission`

### Error: failed to initialize the database: resource temporarily unavailable
This error typically occurs when chain-maind is already running or another process is occupying the required port. To resolve this, check if chain-maind is already active on another port, or consider killing chain-maind then restart it again.   

### How much basecro in 1 CRO?
1 CRO = 10 ^ 8 basecro

### Where can I find API-related documentation?
We regularly update the Chain documentation as required. The API documentation is now available here: https://docs.cronos-pos.org/cronos-pos-integration/chain-integration#api-documentation

### What is the CRO coin type?
The coin type of CRO is 394, and you can refer to the coin type details at
https://github.com/satoshilabs/slips/blob/master/slip-0044.md

### Are rewards and principals separated?
Yes, rewards are separated from your principal staking amount in the sense that the pending reward will be allocated to the account but separated into different parts. 
For example, https://www.mintscan.io/crypto-org/account/cro1u5ryf5jwc2jhd9xyvmasfqzacxp03v8dmly0pc, where we can see that the amounts are separated. 

### What's the reward redemption time?
The reward is allocated to the account every block as mentioned above, the reward can then be redeemed by a "claim-reward" transaction, and the token will be instantly available right after it has been claimed.

&nbsp;

### What is Croeseid testnet?
The Croeseid testnet is the testnet of the Cronos POS Chain which is based on the Cosmos SDK. Detailed information can be found in this [link](https://blog.crypto.com/crypto-com-chain-introduces-croeseid-testnet/).

### What is TCRO?
TCRO is a test token used on the Cronos POS Chain testnet and has no real value. 

### How do I participate in the Croeseid testnet?
Kindly follow this [guide](https://docs.cronos-pos.org/for-node-hosts/getting-started/croeseid-testnet) to join Croeseid testnet as a validator/full node.

### What are the system requirements for running a node in the Croeseid testnet?
To run Cronos PoS Chain nodes in the testnet, you will need a machine with the following minimum requirements to run different types of nodes:

Archive Node (setting pruning = nothing)
- RAM: 16GB (goleveldb)
- Disk: 900GB
- CPU: 4 cores

Default Full Node (setting pruning = default)
- RAM 16GB (goleveldb)
- Disk: 150GB (Depends on how long the node is running for)
- CPU: 4 cores
Refer to [this page](https://docs.cronos-pos.org/for-node-hosts/getting-started/croeseid-testnet) for the latest details.

### How do I get test tokens?
To interact with the blockchain, simply use the [TCRO Faucet](https://cronos-pos.org/faucet) to obtain test CRO tokens for performing transactions on the Croeseid testnet. Note that you will need to [create an address](https://docs.cronos-pos.org/for-users/wallets/cli#keys-management-chain-maind-keys) before using the faucet.

### How long does it take for a node to get back after it is jailed in testnet?
The `downtime_jail_duration` is 3600s (1 hour).

&nbsp;

### Who is running as a validator on Cronos EVM Chain?
Cronos runs on Proof of Authority (POA), a more streamlined and scalable consensus mechanism. Currently, validator hosting is by invitation only. As of now, there are 20+ industry partner validators committed to supporting the Cronos EVM chain.

### How can I become a Cronos EVM validator?
Anyone can run their own Cronos EVM node: see [chain doc](https://docs.cronos.org/for-node-hosts/running-nodes/cronos-mainnet ) for instructions.
However, Cronos EVM Validators are by invitation only at the moment. The application for Cronos EVM validators will not open until further notice. We will make an update once the application opens again. 
Feel free to email contact@cronoslabs.org to register your interest.

### Where can I find RPC URL details for Cronos EVM Chain?
The RPC details are available in this [page](https://docs.cronos.org/for-dapp-developers/chain-integration/public-rpc-endpoints).

### What is the chain ID for Cronos EVM Mainnet?
Ethereum Chain ID: `25`
Cronos Chain ID: `cronosmainnet_25-1`

### How to create a token/project on the Cronos EVM Mainnet? Is there any guide on this?
You can follow this instruction [here](https://docs.cronos.org/for-dapp-developers/cronos-smart-contract).

### What is Cronos EVM Primary Address?
Your eth-form-address begins with `0x… `. *Please make sure you have kept the private key safe and can access the address until any CRO Reward has been distributed and claimed by you.*
You may refer to [this guide](https://docs.cronos.org/for-users/metamask) to create such an address.

### What are the rate limits for the EVM endpoint for Cronos EVM Chain?
Cronos Mainnet: a rate limit is 300 req/min/IP
Cronos Testnet: a rate limit of 500 req/min/IP
If the limit is exceeded, the IP gets blocked for 1 minute. If you need to consistently make more requests than the limits, you may consider setting up your own full node.

### Is WebSocket publicly available for both Cronos EVM chain?
No public WebSocket is available at the moment for EVM chain.

### How can I access the Cronos EVM REST APIs?
To access the Cronos EVM REST APIs for the mainnet, refer to the [API documentation](https://explorer-api-doc.cronos.org/mainnet/) and [register for an account](https://explorer.cronos.org/register) for access. 
To access the Cronos EVM REST APIs for the testnet, refer to the documentation at https://explorer-api-doc.cronos.org/testnet/ and [register for an account](https://explorer.cronos.org/testnet/register) for access.  

###  Are there best practices and security standards that apply to Dapps on Cronos?
The Cronos team strongly encourages all development teams to follow the best practices summarized at this [link](https://docs.cronos.org/for-users/tips-and-faqs).

It will be difficult for us to engage meaningfully with teams who do not publish their code and do not have their smart contracts verified on [Cronos Explorer](https://explorer.cronos.org/).

### If I increase the gas fee, does it help to speed up my transaction?
The current mempool setting works in a first-come-first-serve manner and priority with a gas fee is not supported ( i.e. setting a higher gas fee would not speed up the transaction).

### Why cannot I verify the smart contract on Cronos EVM Testnet with hardhat-etherscan verification?
The hardhat-etherscan verification for Cronos testnet seems not ready yet (on the Etherscan side, not the hardhat side), but eventually will be supported.
For Cronos mainnet, users can refer to https://medium.com/cronos-chain/cronos-developer-series-deploy-verify-your-contracts-using-hardhat-8b6ab6928986.

### How can I request an alternative/paid RPC for Cronos mainnet other than the existing public one?
You may check here for such a service: https://docs.cronos.org/for-dapp-developers/chain-integration/public-rpc-endpoints#commercial-node-providers 

### What are the network details for the Cronos EVM Mainnet?
The Cronos EVM Mainnet has the following details:
- Network Name: Cronos
- New RPC URL:https://evm.cronos.org/ 
- Chain ID: 25 
- Symbol:CRO
- Block explorer URL:https://explorer.cronos.org/ 

The Cronos EVM Testnet has the following details:
- Name: Cronos Testnet
- New RPC URL: https://evm-t3.cronos.org/ 
- Chain ID:338
- Symbol: TCRO
- Block explorer URL: https://explorer.cronos.org/testnet
 
### What is the difference between using the Cronos Play Gaming SDK and Moralis?
In short, there are more similarities than differences, because a lot of this stuff is based on similar open-source libraries. Both are wrappers that allow developers to abstract away the generic read/write functions that you need when you want your user to interact with an Ethereum-compatible chain.

But Moralis SDK for Unity operates at a higher level of abstraction, meaning that it wraps and integrates more functionalities that you would need, as a developer, to code yourself otherwise. For example, Moralis SDK allows you to easily store your user data in their backend servers, whereas the Cronos Play integration of Chainsafe SDK (for Unity) does not include any backend, it's just a bunch of front-end functions that call a blockchain endpoint directly, with nothing in between.

In summary, as a dev, if you want to rely more on "ready-made" libraries and high levels of abstraction on top of basic Ethereum read/write functions, use Moralis. If you want more control over lower-level code and more granular customization of how blockchain transactions are read and written, use the Chainsafe SDK. You can start with Moralis and switch if you see that you need more customization, as switching is not going to require a complete redesign of your code (the underlying logic is similar). 

### Are there best practices and security standards that apply to Dapps on Cronos?**
The Cronos team strongly encourages all development teams to follow the best practices summarized at this [link](https://cronos.org/docs/getting-started/defi-practice.html#introduction).

It will be difficult for us to engage meaningfully with teams who do not publish their code and do not have their smart contracts verified on [Cronos Explorer](https://explorer.cronos.org/).

### Where can I find a list of available developer tools and integration?
Cronos Dev Tools & Integrations List [here](https://docs.cronos.org/for-dapp-developers/dev-tools-and-integrations/overview-of-dev-tools-and-integrations).

### How can I connect my Dapp to commonly used crypto wallets like MetaMask, Crypto.com DeFi Wallet, or Trust Wallet?
See this tutorial: https://medium.com/cronos-chain/cronos-developer-series-connect-your-dapp-with-defi-wallet-metamask-and-trust-wallet-77419fe696a5
Login to the Crypto.com Onchain Wallet can be achieved in three ways: via Wallet Connect (desktop, mobile), via the Onchain Wallet’s browser extension (desktop), or via the Onchain Wallet’s Dapp browser (mobile). See below for how to get your Dapp supported by the Onchain Wallet’s in-app browser.

### Founder's Tips
Visit https://docs.cronos.org/for-dapp-developers/founder-faqs

### How can I get my project listed at Cronos Discover?
Visit https://discover.cronos.org/ where you can add your project to the list of Cronos Discover (or this [Submit](https://airtable.com/appc8zlP6Hctdw7lA/pag8yAWJldMZAmGL9/form) button).

#### How to verify your contracts programmatically on Cronos (using Hardhat)
https://medium.com/cronos-chain/cronos-developer-series-deploy-verify-your-contracts-using-hardhat-8b6ab6928986
https://www.npmjs.com/package/@cronos-labs/hardhat-cronoscan 

#### How to connect your Dapp with DeFi Wallet, MetaMask, and Trust Wallet
https://medium.com/cronos-chain/cronos-developer-series-connect-your-dapp-with-defi-wallet-metamask-and-trust-wallet-77419fe696a5

#### How to create your NFT collection with Solidity, Typescript, and IPFS
https://medium.com/cronos-chain/cronos-developer-series-create-your-nft-collection-with-solidity-typescript-and-ipfs-9b48f90871b9

#### Random Number Generation on Cronos with Witnet
https://medium.com/cronos-chain/random-number-generation-on-cronos-with-witnet-8b871beef59b 

#### Guidelines and tooltips to write secure smart contracts
https://github.com/crytic/building-secure-contracts

 &nbsp;

### How can I access the Cronos zkEVM REST APIs?
To access the Cronos zkEVM REST APIs for the mainnet, refer to the documentation at https://explorer-api-doc.zkevm.cronos.org/mainnet/index.html and register for an API key at [Developer Portal](https://developers.zkevm.cronos.org/register).
To access the Cronos zkEVM REST APIs for the testnet, refer to the documentation at https://explorer-api-doc.zkevm.cronos.org/testnet/ and register for an API key at [Developer Portal](https://developers.zkevm.cronos.org/register).

### What are the network details for the Cronos zkEVM Mainnet?
The Cronos zkEVM Mainnet has the following details:
- Network Name: Cronos zkEVM Mainnet
- RPC URL: https://mainnet.zkevm.cronos.org/ 
- Chain ID: 388
- Currency Symbol: zkCRO
- Block Explorer URL: https://explorer.zkevm.cronos.org/ 
- WebSocket (rate limited): wss://ws.zkevm.cronos.org/

### What are the network details for the Cronos zkEVM Testnet?
The Cronos zkEVM Testnet has the following details:
- Network Name: Cronos zkEVM Testnet
- RPC URL: https://testnet.zkevm.cronos.org/ 
- Chain ID: 240
- Currency Symbol: zkTCRO
- Block Explorer URL: https://explorer.zkevm.cronos.org/testnet/ 
- WebSocket (rate limited): wss://ws.testnet.zkevm.cronos.org

### How can I obtain zkCRO? 
Users can receive zkCRO on Cronos zkEVM by depositing Ethereum CRO on the Cronos zkEVM Bridge. The deposited CRO is staked into the zkCRO contract and mints zkCRO and then bridged to the user’s wallet on Cronos zkEVM. Details are [here](https://docs-zkevm.cronos.org/getting-started/about-zkcro-veth-vusd/zkcro#how-to-obtain-zkcro). Also, zkCRO can be swapped on [H2 Finance](https://h2.finance/). 

### How can I bridge funds from Cronos CRO to zkCRO? 
Several options to bridge funds from Cronos CRO to zkCRO: 
Option 1: 
1) Deposit $CRO (Cronos EVM)
2) Withdraw $CRO (Cronos zkEVM) 
3) Exchange $CRO for $zkCRO on DEX such as [H2 finance](https://h2.finance/swap) 
Note: Gas fees can be paid in CRO on the zkEVM network. 

Option 2: Hold CRO on the Ethereum (obtain CRO by swapping on Uniswap or purchasing from a CEX and withdrawing it to the Ethereum, or get them through [these ways](https://docs.cronos.org/for-users/bridge/app_n_ex)). 
Use [Cronos zkEVM Bridge](https://zkevm.cronos.org/bridge) to swap CRO for zkCRO. 

### How can I bridge funds from LCRO to zkCRO? 
Option 1: Unstake your LCRO on [Veno Finance](https://veno.finance/dashboard/) for CRO (on Cronos). Use [H2 Finance](https://h2.finance/swap) to swap for zkCRO. Please note: Unstaking LCRO takes 28 days. 

Option 2: Sell/Swap your LCRO for CRO on [VVS Finance](https://vvs.finance/limit-orders). Transfer the CRO to a centralized Exchange, withdraw CRO (on the Ethereum network), then use the [Cronos zkEVM Bridge](https://zkevm.cronos.org/bridge/) to get zkCRO. 
Note: we recommend DYOR before using third party projects mentioned above. 

### How can I redeem zkCRO for CRO? 
Given that each zkCRO is backed by LCRO locked on Cronos EVM chain, users can redeem their zkCRO on Cronos zkEVM chain and they will receive CRO on Cronos EVM chain in return.
Steps involved to redeem zkCRO:
- Withdraw zkCRO from Cronos zkEVM to Ethereum
- Claim zkCRO & Redeem for LCRO
- Unstake or sell LCRO for CRO
Users that wish to redeem their zkCRO for CRO can do so via the official [Cronos zkEVM bridge](https://zkevm.cronos.org/bridge). The bridge allows users to withdraw zkCRO for LCRO on Cronos EVM. LCRO can be redeemed for CRO with a 28-32 waiting period. Users that wish to receive their CRO earlier can sell their LCRO within the [Cronos Discover](https://discover.cronos.org/). 

### I claimed ybETH and requested for ETH, but I haven't seen ETH on Etherscan yet. What should I do? 
The bridging process can take more than 24 hours (plus time for the batch to settle). The stated period of "1-10 days" refers to the time required for ybETH to unstake, as it represents staked ETH. 
We recommend checking back later, meanwhile you can also check your transaction status on Etherscan and Cronos zkEVM Explorer to make sure your transactions were successfully performed. We appreciate your patience during this process.  

### How long does it take to bridge from Cronos zkCRO to the Ethereum L1 chain? 
There is a 24hr block timelock execution delay. Each L2 block committed to L1 will have a time lock before it is executed and finalized. After this initial 24hr-period, an additional 24hrs is then needed for your transaction to be included in the next batch and committed to L1, especially during high network traffic. Your funds should be ready to be claimed in 48hrs. If your withdrawal remains stuck after an extended period, please open a support ticket with your account details.

### Is WebSocket publicly available for both Cronos zkEVM chain?
- Mainnet: wss://ws.zkevm.cronos.org;
- Testnet: wss://ws.testnet.zkevm.cronos.org
Rate limits of 300 reqs/min.

&nbsp;

### What is token migration?
Token migration is the process of migrating your CRO (ERC20) on the Ethereum network to CRO on Cronos POS Chain Mainnet. You may read more information [here](https://help.crypto.com/en/articles/5015397-all-about-cro-token-migration).

### Why do I need to migrate and is there a deadline?
The Cronos POS Mainnet is a different blockchain network than the existing CRO (ERC20) on Ethereum. This is the reason why the action of migrating your funds is necessary.
 
For staking CRO on Cronos POS Chain, you need to stake with native CRO. Please note that CRO (ERC20) is required to be used for the DeFi Swap Protocol that’s built on the Ethereum Blockchain and smart contracts, 
 
Currently, there is no deadline for the migration. We highly recommend the existing CRO holders migrate as soon as possible to enjoy the benefits of the Cronos POS Chain and also stay tuned to our news channel for the latest updates.
 
### How do I migrate my ERC20 CRO to the native CRO on DeFi Wallet? Which migration method is most suitable for me?
There will be multiple ways available for you to migrate CRO to the Mainnet:

- **Through Crypto.com App and Exchange:**
  Suitable for CRO holders who prefer an easy way to perform the migration. Once you have registered an account, you can deposit your CRO (ERC20) from an external wallet and then withdraw Cronos POS Coin back to your external wallet.
- **Through Crypto.com Onchain wallet:** 
  Suitable for non-custodial CRO holders. It is one of the easiest ways to perform migration for non-custodial wallet users. Our in-app CRO Migration Tool allows you to easily convert your ERC20 CRO to native CRO. Native CRO is the native currency of the Cronos POS Chain, which enables near-instant CRO transactions with minimal fees. 
- **Through our featured exchanges:**
  Suitable for CRO holders who have previously registered on these exchanges. Please follow the instructions on the specific exchanges for more details.
- **Through Cronos POS Chain [Token Migration Web Tool](https://docs.cronos-pos.org/for-users/token-migration):**
  Suitable for advanced non-custodial wallet users (support MetaMask or Ledger wallet connection). This process involves more complicated steps to perform the migration manually as you might need to generate the native CRO wallet address, so it is recommended for advanced users only.

 
 &nbsp;

## Useful links
Cronos POS
[Website](https://cronos-pos.org/)
[Chain Documentation](https://docs.cronos-pos.org)
[Explorer](https://cronos-pos.org/explorer)
[API](https://docs.cronos-pos.org/cronos-pos-integration/chain-integration)
[Genesis File and Seed Nodes](https://github.com/crypto-org-chain/mainnet)
[Faucet](https://cronos-pos.org/faucet)
[Substack Blog Post](https://blog.cronos-pos.org/)
[Medium Blog Post](https://medium.com/crypto-org-chain)
[Developer community Discord](https://discord.com/invite/5JTk2ppsY3)
[Telegram Announcement Channel](https://t.me/Cronos_Announcements) 

Cronos EVM
[Website](https://cronos.org/)
[Cronos Labs](https://www.cronoslabs.org/) 
[Cronos Explorer](https://explorer.cronos.org/)  
[Chain Documentation](https://docs.cronos.org/)
[Binaries](https://github.com/crypto-org-chain/cronos/releases)
[API](https://explorer-api-doc.cronos.org/mainnet/)
[Dev Tools](https://docs.cronos.org/for-dapp-developers/useful-projects-on-cronos/overview-of-dev-tools-and-integrations)
[Faucet](https://cronos.org/faucet)
 
Cronos zkEVM
[Chain Doc](https://docs-zkevm.cronos.org/)
[Cronos zkEVM Explorer](https://explorer.zkevm.cronos.org/testnet/)
[Faucet](https://zkevm.cronos.org/faucet)
[Cronos zkEVM Developers Portal](https://developers.zkevm.cronos.org/)

Others
[Cronos Discover](https://discover.cronos.org/)
[Roadmap](https://medium.com/cronos-chain/cronos-chain-journey-so-far-and-2022-roadmap-3475abbe6254)
[Whitepaper](https://whitepaper.cronos.org/)
[Cronos Labs](https://www.cronoslabs.org/)
[Snapshot](https://snapshot.cronos.org/)
[Cronos EVM Developer Resources](https://cronos.org/build)
[Blog Post](https://blog.cronos.org/s/cronos-developer-series)
[Cronos Bounty](https://hackenproof.com/company/cronos/programs)
[Youtube](https://www.youtube.com/@cronos_chain)
[Contact Form](https://airtable.com/app5F2H2Qp46Q9fGC/shr2pGmQR5lzoMOBh)

**Comment by varakutakrak** [comment]


**Reply by varakutakrak** [reply]
1 cro
